### PR TITLE
docs: Clarify load balancing policies

### PR DESCRIPTION
**Pull request by brawer**
I’m new to Envoy and impressed by the nicely written introduction. However, the page about [Load Balancing Policies](https://www.envoyproxy.io/docs/envoy/v1.25.1/intro/arch_overview/upstream/load_balancing_policies) seems a little confusing. Here’s an attempt to make it more legible; please double-check whether my changes actually make sense.

Risk Level: Low

### CODE REVIEW

**Comment by phlax** [comment]
could you fix DCO please @brawer in this case you will need to force push to fix - you may want to check this section which has info on automating that https://github.com/envoyproxy/envoy/blob/main/CONTRIBUTING.md#submitting-a-pr

(top tip to only use the DCO hook `echo NO_VERIFY=1 >> .env`)

File: docs/root/intro/arch_overview/upstream/load_balancing_policies.rst [hunk 1]
```diff
@@ -1,20 +1,19 @@
...
+Extendable load balancing policies can be
+:ref:`configured <envoy_v3_api_field_config.cluster.v3.Cluster.load_balancing_policy>` separately for each cluster, also by calling
+:ref:`APIs <envoy_v3_api_config_load_balancer_policies>`.
 
+Developers can implement custom, configurable policies in C++.
 
-See built-in load balancing policies by :ref:`APIs <envoy_v3_api_config_load_balancer_policies>`.
+In the past, Envoy used an
+:ref:`enum <envoy_v3_api_field_config.cluster.v3.Cluster.lb_policy>`
+to specify load balancing policies. This enum is still supported for
```
Comment (phlax) [inline docs/root/intro/arch_overview/upstream/load_balancing_policies.rst:12]:
```suggestion
to specify load balancing policies. This ``enum`` is still supported for
```

File: docs/root/intro/arch_overview/upstream/load_balancing_policies.rst [hunk 2]
```diff
@@ -1,20 +1,19 @@
...
+backward compatibility, but deprecated; extendable load balancing policies
```
Comment (phlax) [inline docs/root/intro/arch_overview/upstream/load_balancing_policies.rst:13]:
```suggestion
backward compatibility, but deprecated; ref:`extendable load balancing policies <envoy_v3_api_config_load_balancer_policies>`
```

File: docs/root/intro/arch_overview/upstream/load_balancing_policies.rst [hunk 3]
```diff
@@ -1,20 +1,19 @@
...
-:ref:`load balancer policy <envoy_v3_api_field_config.cluster.v3.Cluster.load_balancing_policy>`.
-And the developer can implement a custom load balancing policy and configured it.
+Extendable load balancing policies can be
+:ref:`configured <envoy_v3_api_field_config.cluster.v3.Cluster.load_balancing_policy>` separately for each cluster, also by calling
+:ref:`APIs <envoy_v3_api_config_load_balancer_policies>`.
 
+Developers can implement custom, configurable policies in C++.
 
-See built-in load balancing policies by :ref:`APIs <envoy_v3_api_config_load_balancer_policies>`.
+In the past, Envoy used an
```
Comment (phlax) [inline docs/root/intro/arch_overview/upstream/load_balancing_policies.rst:10]: maybe we could use a 

```rst

.. note::
   ....

```

here

**Review by phlax — COMMENTED** [review]
thanks for working on this @brawer 

docs updates are very much appreciated

**Comment by brawer** [comment]
> could you fix DCO

Done.

File: docs/root/intro/arch_overview/upstream/load_balancing_policies.rst (hunk 3 above)
Comment (brawer) [reply docs/root/intro/arch_overview/upstream/load_balancing_policies.rst]: Done.

**Review by phlax — APPROVED** [review]
lgtm, thanks @brawer

File: docs/root/intro/arch_overview/upstream/load_balancing_policies.rst [hunk 4]
```diff
@@ -1,20 +1,21 @@
...
 
+Developers can implement custom, configurable policies in C++.
 
-See built-in load balancing policies by :ref:`APIs <envoy_v3_api_config_load_balancer_policies>`.
+.. note::
+  In the past, Envoy used an
+  :ref:`enum <envoy_v3_api_field_config.cluster.v3.Cluster.lb_policy>`
+  to specify load balancing policies. This ``enum`` is still supported for
+  backward compatibility, but deprecated;
+  ref:`extendable load balancing policies <envoy_v3_api_config_load_balancer_policies>`
```
Comment (phlax) [inline docs/root/intro/arch_overview/upstream/load_balancing_policies.rst:15]:
```suggestion
  :ref:`extendable load balancing policies <envoy_v3_api_config_load_balancer_policies>`
```

apologies - spotted a rendering issue at the last minute

Comment (phlax) [reply docs/root/intro/arch_overview/upstream/load_balancing_policies.rst:15]: also - while you are there - could i suggest we put this to a separate line and replace `;` with `.` after "...deprecated."

i think this is the important bit

you will need a single blank line to render it correctly in rst

**Pull request merged**